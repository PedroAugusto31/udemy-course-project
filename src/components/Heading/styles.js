import styled, { css } from "styled-components";

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.xhuge};
  `,
};

export const Title = styled.h1`
  ${({ theme, isLight, size }) => css`
    color: ${isLight ? theme.colors.blackTheme : theme.colors.mainColors};
    ${titleSize[size](theme)};
  `}
`;
