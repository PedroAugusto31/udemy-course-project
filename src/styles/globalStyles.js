import { createGlobalStyle, css } from "styled-components";

export const GlobalStyles = createGlobalStyle`

 *{
   margin: 0;
   padding: 0;
   box-sizing: border-box;
 }

 html {
   font-size: 62.5%;
   scroll-behavior: smooth;
 }

 body {
   padding: 0;
   font-size: 1.6rem;
   font-family: 'Open Sans', sans-serif;
 }

 h1, h2, h3, h4, h5, h6 {
   font-family: 'Montserrat', sans-serif;
   font-weight: 900;
   margin: ${({ theme }) => theme.spacings.large} 0;
 }

 p {
   margin: ${({ theme }) => theme.spacings.medium} 0;
 }

 ul, ol {
   margin: ${({ theme }) => theme.spacings.medium};
   padding: ${({ theme }) => theme.spacings.medium};
 }

 a {
  color: ${({ theme }) => theme.colors.mainColor};
 }

 .table {
   width: 100%;
   overflow-y: auto;
 }
`;
