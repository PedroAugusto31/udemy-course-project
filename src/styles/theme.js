export const theme = {
  colors: {
    mainBg: "#ffffff",
    blackTheme: "#323232",
  },
};
