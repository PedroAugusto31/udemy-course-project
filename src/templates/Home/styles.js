import styled, { css } from "styled-components";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.secondary};
  `}
  font-style: italic;
`;
