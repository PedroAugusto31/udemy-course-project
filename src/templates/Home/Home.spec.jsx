import { screen } from "@testing-library/react";
import { Home } from ".";
import { theme } from "../../styles/theme";
import { renderTheme } from "../../utils/renderTheme";

describe("<Home />", () => {
  test("renders learn react link", () => {
    renderTheme(<Home />);
    //render(<Home />);
    // debug();

    const title = screen.getByRole("heading", { name: "hello" }).parentElement;
    expect(title).toHaveStyle({ background: theme.colors.blackTheme });
    expect(title).toMatchSnapshot();
  });
});
