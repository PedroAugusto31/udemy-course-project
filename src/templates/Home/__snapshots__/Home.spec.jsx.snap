// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders learn react link 1`] = `
<div
  class="sc-bdvvtL jFRBlj"
>
  <h1>
    hello
  </h1>
</div>
`;
